.github/workflows/ci.yml

```diff
@@ -0,0 +1,38 @@
+name: Continuous Integration
+concurrency: 
+    group: ${{ github.workflow }}-${{ github.ref }}
+    cancel-in-progress: true
+
+on:
+  push:
+    branches: '**'
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        submodules: recursive
+        fetch-depth: 0
+    - name: Set up JDKs ☕
+      uses: actions/setup-java@v3
+      with:
+        java-version: |
+          8
```

Comment:
Does it mean, ecj will run on Java 8? That is not possible.

Reply:
No, it just makes Java 8 will be available on the CI machine.

Reply:
... and why do we need Java 8 on CI machine?

Reply:
Tycho requires Java 8 to properly resolve Java 8 APIs. It can be turned off and the current Java runtime can be used instead (Java 17), but we've seen cases where some API disappear in newer versions and thus the code targetting Java 8 that cannot be compiled any longer by Java 17.
IIRC, Java 8 is also installed on Jenkins and configured in java toolchains.

Reply:
> the code targetting Java 8 that cannot be compiled any longer by Java 17.

We do not have code targeting Java 8 AFAIK, and annotations don't use any Java 8 specific API's, so why all this overhead adding something we don't need?

Reply:
You need to ensure compatibility with Java 8 if you target Java 8 (in https://github.com/eclipse-jdt/eclipse.jdt.core/blob/master/org.eclipse.jdt.annotation_v1/META-INF/MANIFEST.MF#L8 ). Compatibility cannot be verified at build-time without a Java 8 installation being visible to Tycho.
The build config here is similar to Jenkins one. I wouldn't make it a requirement to chnage the build config to something too different from "main" build.
If you wish to get rid of the requirement of having Java 8 installed for build to succeed, please open another issue; it can be changed from pom.xml, but this would also affect the Jenkins build. It's another topic.